tools/testsys/src/vmware_k8s.rs

```diff
@@ -0,0 +1,242 @@
+            crd_input.arch,
+            crd_input
+                .starting_version
+                .as_ref()
+                .context(error::InvalidSnafu {
+                    what: "The starting version must be provided for migration testing"
+                })?
+        ))
+    }
+
+    /// Create an vSphere K8s cluster CRD with the `cluster_name` in `cluster_input`.
+    async fn cluster_crd<'a>(&self, cluster_input: ClusterInput<'a>) -> Result<CreateCrdOutput> {
+        let control_plane_endpoint = cluster_input
+            .crd_input
+            .config
+            .control_plane_endpoint
+            .as_ref()
+            .context(error::InvalidSnafu {
+                what: "The control plane endpoint is required for vmware cluster creation.",
+            })?;
+        let labels = cluster_input.crd_input.labels(btreemap! {
+            "testsys/type".to_string() => "cluster".to_string(),
+            "testsys/cluster".to_string() => cluster_input.cluster_name.to_string(),
+            "testsys/controlPlaneEndpoint".to_string() => control_plane_endpoint.to_string(),
+        });
+
+        // Check if the cluster already has a CRD
+        if let Some(cluster_crd) = cluster_input
+            .crd_input
+            .existing_crds(
+                &labels,
+                &[
+                    "testsys/cluster",
+                    "testsys/type",
+                    "testsys/controlPlaneEndpoint",
+                ],
+            )
+            .await?
+            .pop()
+        {
+            return Ok(CreateCrdOutput::ExistingCrd(cluster_crd));
+        }
+
+        // Check if an existing cluster is using this endpoint
+        let existing_clusters = cluster_input
+            .crd_input
+            .existing_crds(&labels, &["testsys/type", "testsys/controlPlaneEndpoint"])
+            .await?;
+
+        let cluster_version =
+            K8sVersion::from_str(cluster_input.crd_input.variant.version().context(
+                error::MissingSnafu {
+                    item: "K8s version".to_string(),
+                    what: "aws-k8s variant".to_string(),
+                },
+            )?)
+            .map_err(|_| error::Error::K8sVersion {
+                version: cluster_input.crd_input.variant.to_string(),
+            })?;
+
+        let vsphere_k8s_crd = VSphereK8sClusterConfig::builder()
+            .name(cluster_input.cluster_name)
+            .set_labels(Some(labels))
+            .control_plane_endpoint_ip(control_plane_endpoint)
+            .creation_policy(CreationPolicy::IfNotExists)
+            .version(cluster_version)
+            .ova_name(self.image_id(cluster_input.crd_input)?)
+            .tuf_repo(
+                cluster_input
+                    .crd_input
+                    .tuf_metadata()
+                    .context(error::InvalidSnafu {
+                        what: "TUF repo information is required for vmware cluster creation.",
+                    })?,
```

Comment:
It seems like we're precluding the use of long-lived test cluster fixtures here.

For parity with EKS on AWS, I would expect to create a cluster control plane using the "production" builds from EKS-A, and then only use local Bottlerocket builds for the worker nodes.

Reply:
Since these clusters are easier to create and destroy, we do not have them persisting (in order to preserve control-plane-ips)

Reply:
I'd like to backlog an issue for this at least - if it turns out not to be needed, we can close the issue, but the inconsistency across platforms bothers me.